lib/src/rules/use_super_initializers.dart

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Use super-initializer parameters where possible.';
+
+const _details = r'''
+"Forwarding constructor"s, that do nothing except forward parameters to their 
+superclass constructors should take advantage of super-initializer parameters 
+rather than repeating the names of parameters when passing them to the 
+superclass constructors.  This makes the code more concise and easier to read
+and maintain.
+
+**DO** use super-initializer parameters where possible.
+
+**BAD:**
+```dart
+class A {
+  A({int? x, int? y});
+}
+class B extends A {
+  B({int? x, int? y}) : super(x: x, y: y);
+}
+```
+
+**GOOD:**
+```dart
+class A {
+  A({int? x, int? y});
+}
+class B extends A {
+  B({super.x, int? y}) : super(y: y);
+}
+```
+''';
+
+class UseSuperInitializers extends LintRule {
+  UseSuperInitializers()
+      : super(
+            name: 'use_super_initializers',
+            description: _desc,
+            details: _details,
+            maturity: Maturity.experimental,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    if (!context.isEnabled(Feature.super_parameters)) return;
+
+    var visitor = _Visitor(this, context);
+    registry.addConstructorDeclaration(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LinterContext context;
+  final LintRule rule;
+
+  _Visitor(this.rule, this.context);
+
+  void check(
+      SuperConstructorInvocation initializer, FormalParameterList parameters) {
+    var constructorElement = initializer.staticElement;
+    if (constructorElement == null) return;
+
+    for (var parameter in parameters.parameters) {
+      var parameterElement = parameter.declaredElement;
+      if (parameterElement == null) continue;
+
+      // todo(pq): consolidate logic shared w/ server (https://github.com/dart-lang/linter/issues/3263)
+      if (parameterElement.isPositional) {
+        if (_checkPositionalParameter(
+            parameter, parameterElement, constructorElement, initializer)) {
+          rule.reportLint(initializer);
+        }
+      } else if (parameterElement.isNamed) {
+        if (_checkNamedParameter(
+            parameter, parameterElement, constructorElement, initializer)) {
+          rule.reportLint(initializer);
```

Comment:
Given that we're suggesting that the parameter be changed, I think it would make more sense to highlight the name of the `parameter` rather than the `initializer`.